### DISCUSSION TITLE
Weird edge case with upgrading and using the better wifi app.

### DISCUSSION THREAD

**Discussion by Grenrose** [OnionUI/Onion#1075] [General]
Running the first RC of 4.2.0 on my mini+ I chose to use the better wifi app because of the special characters in my wifi password, and got it all set up and working fine. I went and checked the update time and OTA updates on start up settings. When I upgraded to the current RC snap shot onionOS ended up getting stuck on boot because it was trying to endlessly reach the time server. I waited a good 10 minutes, expecting it to eventually timeout and boot but it just never did. I tried deleting wpa supplicant thinking that the app was probably just automating the editing of it but that did not fix it. I eventually just had to copy everything off and reflash the SD card to a stock onion install and then copy back over everything not system related. Basically wasn't sure what else it might have edited or what files to try to get rid of.

I'm not knowledgeable enough and I didn't do enough testing to try to narrow it down so I apologize I was kinda focused on just getting it back to working. Also I know this possibly isn't an onionOS specific thing so I don't expect you to fix it, but it might be useful to state that using the app may end up breaking your install on update if you select those options. Maybe you could if possible just make a timeout for those options, that would in a roundabout way fix it. Although not sure how much effort that would take,

**Comment by thomasklemenc** [comment]
Fixed by #1073 

If you need a workaround, put your SD card into a computer and delete the file `SD:/.tmp_update/config/.ntpWait`

**Comment by XK9274** [comment]
This isn't related to my app - it was a loop without an exit condition in the NTP logic being addressed here: https://github.com/OnionUI/Onion/pull/1073. 

Betterwifi doesn't do anything at all to Onion, it only interfaces with wpa_supplicant.conf and wpa_cli which are firmware level.

When you flashed the SD card you removed all the NTP config states so you likely bypassed this loop.

**Comment by Grenrose** [comment]
AH cool, well damn now i wish i would have seen that. That was very helpful, thank you both :D.
